openpype/hosts/nuke/plugins/publish/collect_writes.py

```diff
@@ -69,11 +69,14 @@ def _set_existing_files_data(self, instance, colorspace):
             instance, collected_frames
         )
 
-        # inject colorspace data
-        self.set_representation_colorspace(
-            representation, instance.context,
-            colorspace=colorspace
-        )
+        # To avoid errors only try injecting colorspace
+        # if any existing files were found on disk
+        if representation["files"]:
+            # inject colorspace data
+            self.set_representation_colorspace(
```

Comment:
shouldn't the fix by on the set_representation_colorspace instead so if there's no files it doesn't try to set the colorspace? although one could argue that the filename requirement of set_representation_colorspace is also unnecessary, that should only be for the case when rules are applied but it could be coming as an arg directly

Reply:
```python
    # get one filename
    filename = representation["files"]
    if isinstance(filename, list):
        filename = filename[0]

    # get matching colorspace from rules
    colorspace = colorspace or get_imageio_colorspace_from_filepath(
        filename, host_name, project_name,
        config_data=config_data,
        file_rules=file_rules,
        project_settings=project_settings
    )
```
should just be:
```python
if representation["files"]:
    filename = representation["files"]
    if isinstance(filename, list):
        filename = filename[0]

    # get matching colorspace from rules
    colorspace = colorspace or get_imageio_colorspace_from_filepath(
        filename, host_name, project_name,
        config_data=config_data,
        file_rules=file_rules,
        project_settings=project_settings
    )
```
Although if I'm passing `colorspace` as an arg.. I'd expect that to take priority over the file rules so maybe it should also change:
```python
# if no colorspace is passed, try to infer it from the file rules
if not colorspace and representation["files"]:
    filename = representation["files"]
    if isinstance(filename, list):
        filename = filename[0]

    # get matching colorspace from rules
    colorspace = get_imageio_colorspace_from_filepath(
        filename, host_name, project_name,
        config_data=config_data,
        file_rules=file_rules,
        project_settings=project_settings
    )
```

Reply:
@tokejepsen please address this, this same issue will happen on any other plugins that make use of the same function, this is not something particular to Nuke

Reply:
This needs resolving before further testing.